File: app/src/main/java/com/example/coursework_3course_budgetapp/Operation.java
package com.example.coursework_3course_budgetapp;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.io.Serializable;

@Entity
public class Operation implements Serializable {
    @PrimaryKey(autoGenerate = true)
    private int id;

    @ColumnInfo(name = "Title")
    private String title;

    @ColumnInfo(name = "MoneyAmmount")
    private float ammount;

    @ColumnInfo(name = "Type")
    private String Type;

    @ColumnInfo(name = "Time")
    private int time;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public float getAmmount() {
        return ammount;
    }

    public void setAmmount(float ammount) {
        this.ammount = ammount;
    }

    public String getType() {
        return Type;
    }

    public void setType(String type) {
        Type = type;
    }

    public int getTime() {
        return time;
    }

    public void setTime(int time) {
        this.time = time;
    }
}

File: app/src/main/java/com/example/coursework_3course_budgetapp/Operation.kt
package com.example.coursework_3course_budgetapp

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey

import androidx.room.PrimaryKey
import java.io.Serializable


@Entity(tableName = "Operations",
 foreignKeys = [ForeignKey(entity = Category::class,
     parentColumns = ["id"],
     childColumns = ["categoryID"],
     onDelete = ForeignKey.CASCADE),
     ForeignKey(entity = Bill::class,
         parentColumns = ["id"],
         childColumns = ["billID"],
         onDelete = ForeignKey.CASCADE)])

class Operation : Serializable {
    @PrimaryKey(autoGenerate = true)
    var id = 0

    @ColumnInfo(name = "CategoryID")
    var categoryID: Int? = null

    @ColumnInfo(name = "MoneyAmmount")
    var ammount = 0f

    @ColumnInfo(name = "Type")
    var type: String? = null

    @ColumnInfo(name = "Count")
    var count = 0

    @ColumnInfo(name = "Description")
    var description: String? = null

    @ColumnInfo(name = "BillID")
    var accountID: Int? = null
}